hypre@2.25.0 +rocm fails: _hypre_utilities.hpp:116:10: fatal error: 'rocsparse.h' file not found

### Steps to reproduce the issue

Reproducible using:
* `spack@develop` (4c2f357f0e8c5a2ce5609e654d5bfc435b6caee5 from `Mon Aug 15 18:00:33 2022 -0700`)
* `ecpe4s/ubuntu22.04-runner-x86_64:2022-08-01`
* `%gcc@11.2.0`
* ROCm 5.2.0

Concrete spec: [hypre.spec.json.txt](https://github.com/spack/spack/files/9354617/hypre.spec.json.txt)

<details><summary>Concretization</summary>
<pre>
  -   hypre@2.25.0%gcc@11.2.0~complex~cuda~debug+fortran~gptune~int64~internal-superlu~mixedint+mpi~openmp+rocm+shared~superlu-dist~unified-memory amdgpu_target=gfx90a arch=linux-ubuntu22.04-x86_64
[+]      ^hip@5.2.0%gcc@11.2.0~ipo build_type=Release patches=959d1fe arch=linux-ubuntu22.04-x86_64
[+]          ^cmake@3.23.3%gcc@11.2.0~doc+ncurses+ownlibs~qt build_type=Release arch=linux-ubuntu22.04-x86_64
[+]              ^ncurses@6.2%gcc@11.2.0~symlinks+termlib abi=none arch=linux-ubuntu22.04-x86_64
[+]                  ^pkgconf@1.8.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]              ^openssl@1.1.1q%gcc@11.2.0~docs~shared certs=mozilla patches=3fdcf2d arch=linux-ubuntu22.04-x86_64
[+]                  ^ca-certificates-mozilla@2022-07-19%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^perl@5.34.1%gcc@11.2.0+cpanm+shared+threads arch=linux-ubuntu22.04-x86_64
[+]                      ^berkeley-db@18.1.40%gcc@11.2.0+cxx~docs+stl patches=b231fcc arch=linux-ubuntu22.04-x86_64
[+]                      ^bzip2@1.0.8%gcc@11.2.0~debug~pic+shared arch=linux-ubuntu22.04-x86_64
[+]                          ^diffutils@3.8%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                              ^libiconv@1.16%gcc@11.2.0 libs=shared,static arch=linux-ubuntu22.04-x86_64
[+]                      ^gdbm@1.19%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                          ^readline@8.1.2%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^zlib@1.2.12%gcc@11.2.0+optimize+pic+shared patches=0d38234 arch=linux-ubuntu22.04-x86_64
[+]          ^comgr@5.2.0%gcc@11.2.0~ipo build_type=Release arch=linux-ubuntu22.04-x86_64
[+]              ^llvm-amdgpu@5.2.0%gcc@11.2.0~ipo~link_llvm_dylib~llvm_dylib~openmp+rocm-device-libs build_type=Release patches=a08bbe1 arch=linux-ubuntu22.04-x86_64
[+]                  ^python@3.8.13%gcc@11.2.0+bz2+ctypes+dbm~debug+ensurepip+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93,4c24573,f2fd060 arch=linux-ubuntu22.04-x86_64
[+]                      ^expat@2.4.8%gcc@11.2.0+libbsd arch=linux-ubuntu22.04-x86_64
[+]                          ^libbsd@0.11.5%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                              ^libmd@1.0.4%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^gettext@0.21%gcc@11.2.0+bzip2+curses+git~libunistring+libxml2+tar+xz arch=linux-ubuntu22.04-x86_64
[+]                          ^libxml2@2.9.13%gcc@11.2.0~python arch=linux-ubuntu22.04-x86_64
[+]                              ^xz@5.2.5%gcc@11.2.0+pic libs=shared,static arch=linux-ubuntu22.04-x86_64
[+]                          ^tar@1.34%gcc@11.2.0 zip=pigz arch=linux-ubuntu22.04-x86_64
[+]                              ^pigz@2.7%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                              ^zstd@1.5.2%gcc@11.2.0+programs compression=none libs=shared,static arch=linux-ubuntu22.04-x86_64
[+]                      ^libffi@3.4.2%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^sqlite@3.38.5%gcc@11.2.0+column_metadata+dynamic_extensions+fts~functions+rtree arch=linux-ubuntu22.04-x86_64
[+]                      ^util-linux-uuid@2.37.4%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^z3@4.8.16%gcc@11.2.0~gmp~ipo~python build_type=RelWithDebInfo arch=linux-ubuntu22.04-x86_64
[+]              ^rocm-cmake@5.2.0%gcc@11.2.0~ipo build_type=Release arch=linux-ubuntu22.04-x86_64
[+]          ^glx@1.4%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]              ^mesa@22.1.2%gcc@11.2.0+glx+llvm+opengl~opengles+osmesa~strip buildtype=release default_library=shared patches=ada85be arch=linux-ubuntu22.04-x86_64
[+]                  ^bison@3.8.2%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^m4@1.4.19%gcc@11.2.0+sigsegv patches=9dc5fbd,bfdffa7 arch=linux-ubuntu22.04-x86_64
[+]                          ^libsigsegv@2.13%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^flex@2.6.3%gcc@11.2.0+lex~nls arch=linux-ubuntu22.04-x86_64
[+]                      ^findutils@4.9.0%gcc@11.2.0 patches=440b954 arch=linux-ubuntu22.04-x86_64
[+]                  ^glproto@1.4.17%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^util-macros@1.19.3%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^libunwind@1.6.2%gcc@11.2.0~block_signals~conservative_checks~cxx_exceptions~debug~debug_frame+docs+pic+tests+weak_backtrace+xz~zlib components=none libs=shared,static arch=linux-ubuntu22.04-x86_64
[+]                  ^libx11@1.7.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^inputproto@2.3.2%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^kbproto@1.0.7%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^libxcb@1.14%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                          ^libpthread-stubs@0.4%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                          ^libxau@1.0.8%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                              ^xproto@7.0.31%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                          ^libxdmcp@1.1.2%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                          ^xcb-proto@1.14.1%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^xextproto@7.3.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^xtrans@1.3.5%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^libxext@1.3.3%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^libxt@1.1.5%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^libice@1.0.9%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^libsm@1.2.3%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^meson@0.62.2%gcc@11.2.0 patches=9c87472,aa6c50d arch=linux-ubuntu22.04-x86_64
[+]                      ^ninja@1.11.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^py-pip@22.1.2%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^py-setuptools@63.0.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^py-wheel@0.37.1%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^py-mako@1.1.6%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^py-markupsafe@2.0.1%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^xrandr@1.5.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                      ^libxrandr@1.5.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                          ^libxrender@0.9.10%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                              ^renderproto@0.11.1%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                          ^randrproto@1.5.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]          ^hsa-rocr-dev@5.2.0%gcc@11.2.0+image~ipo+shared build_type=Release patches=71e6851 arch=linux-ubuntu22.04-x86_64
[+]              ^elfutils@0.186%gcc@11.2.0+bzip2~debuginfod~nls+xz arch=linux-ubuntu22.04-x86_64
[+]              ^hsakmt-roct@5.2.0%gcc@11.2.0~ipo+shared build_type=Release patches=f926273 arch=linux-ubuntu22.04-x86_64
[+]                  ^libdrm@2.4.110%gcc@11.2.0~docs arch=linux-ubuntu22.04-x86_64
[+]                      ^libpciaccess@0.16%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                          ^libtool@2.4.7%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]                  ^numactl@2.0.14%gcc@11.2.0 patches=4e1d78c,62fc8a8,ff37630 arch=linux-ubuntu22.04-x86_64
[+]                      ^autoconf@2.69%gcc@11.2.0 patches=35c4492,7793209,a49dd5b arch=linux-ubuntu22.04-x86_64
[+]                      ^automake@1.16.5%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]              ^xxd-standalone@8.2.1201%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]          ^perl-file-which@1.22%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]          ^perl-uri-encode@1.1.1%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]              ^perl-module-build@0.4224%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]          ^rocminfo@5.2.0%gcc@11.2.0~ipo build_type=Release arch=linux-ubuntu22.04-x86_64
[+]          ^roctracer-dev-api@5.2.0%gcc@11.2.0 arch=linux-ubuntu22.04-x86_64
[+]      ^mpich@4.0.2%gcc@11.2.0~argobots~cuda+fortran+hwloc+hydra+libxml2+pci~rocm+romio~slurm~two_level_namespace~vci~verbs~wrapperrpath datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu22.04-x86_64
[+]          ^hwloc@2.8.0%gcc@11.2.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm+shared arch=linux-ubuntu22.04-x86_64
[+]          ^libfabric@1.14.1%gcc@11.2.0~debug~kdreg fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu22.04-x86_64
[+]          ^yaksa@0.2%gcc@11.2.0~cuda~rocm arch=linux-ubuntu22.04-x86_64
[+]      ^openblas@0.3.20%gcc@11.2.0~bignuma~consistent_fpcsr~ilp64+locking+pic+shared symbol_suffix=none threads=openmp arch=linux-ubuntu22.04-x86_64
</pre></details>

Install error:
```
$> spack install -f ./hypre.spec.json
...
==> Installing hypre-2.25.0-6rsc2fdt7na5sqleojg3uug6klohx2f6
==> No binary for hypre-2.25.0-6rsc2fdt7na5sqleojg3uug6klohx2f6 found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/archive/f9/f9fc8371d91239fca694284dab17175bfda3821d7b7a871fd2e8f9d5930f303c.tar.gz
==> No patches needed for hypre
==> hypre: Executing phase: 'autoreconf'
==> hypre: Executing phase: 'configure'
==> hypre: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16'

11 errors found in build log:
     186    hipcc --amdgpu-target=gfx90a -fPIC -O2 -x hip -std=c++14   -DHAVE_CONFIG_H -I.. -I./.. -I./../struct_mv -I.   -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-ow
            pe5je3fhes7ivmri3nbrw6l44qe3ak/rocthrust/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocprim/include -I/spack/opt/s
            pack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocsparse/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fh
            es7ivmri3nbrw6l44qe3ak/rocrand/include      -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/mpich-4.0.2-gpzf75yhrekurnz3v63fmevfjez2x2q2/include  -c handle.c -o handle.ob
            j
     187    hipcc --amdgpu-target=gfx90a -fPIC -O2 -x hip -std=c++14   -DHAVE_CONFIG_H -I.. -I./.. -I./../struct_mv -I.   -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-ow
            pe5je3fhes7ivmri3nbrw6l44qe3ak/rocthrust/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocprim/include -I/spack/opt/s
            pack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocsparse/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fh
            es7ivmri3nbrw6l44qe3ak/rocrand/include      -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/mpich-4.0.2-gpzf75yhrekurnz3v63fmevfjez2x2q2/include  -c int_array.c -o int_ar
            ray.obj
     188    hipcc --amdgpu-target=gfx90a -fPIC -O2 -x hip -std=c++14   -DHAVE_CONFIG_H -I.. -I./.. -I./../struct_mv -I.   -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-ow
            pe5je3fhes7ivmri3nbrw6l44qe3ak/rocthrust/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocprim/include -I/spack/opt/s
            pack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocsparse/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fh
            es7ivmri3nbrw6l44qe3ak/rocrand/include      -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/mpich-4.0.2-gpzf75yhrekurnz3v63fmevfjez2x2q2/include  -c memory.c -o memory.ob
            j
     189    hipcc --amdgpu-target=gfx90a -fPIC -O2 -x hip -std=c++14   -DHAVE_CONFIG_H -I.. -I./.. -I./../struct_mv -I.   -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-ow
            pe5je3fhes7ivmri3nbrw6l44qe3ak/rocthrust/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocprim/include -I/spack/opt/s
            pack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocsparse/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fh
            es7ivmri3nbrw6l44qe3ak/rocrand/include      -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/mpich-4.0.2-gpzf75yhrekurnz3v63fmevfjez2x2q2/include  -c omp_device.c -o omp_d
            evice.obj
     190    hipcc --amdgpu-target=gfx90a -fPIC -O2 -x hip -std=c++14   -DHAVE_CONFIG_H -I.. -I./.. -I./../struct_mv -I.   -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-ow
            pe5je3fhes7ivmri3nbrw6l44qe3ak/rocthrust/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocprim/include -I/spack/opt/s
            pack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fhes7ivmri3nbrw6l44qe3ak/rocsparse/include -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/hip-5.2.0-owpe5je3fh
            es7ivmri3nbrw6l44qe3ak/rocrand/include      -I/spack/opt/spack/linux-ubuntu22.04-x86_64/gcc-11.2.0/mpich-4.0.2-gpzf75yhrekurnz3v63fmevfjez2x2q2/include  -c nvtx.c -o nvtx.obj
     191    In file included from device_utils.c:10:
  >> 192    ./_hypre_utilities.hpp:116:10: fatal error: 'rocsparse.h' file not found
     193    #include <rocsparse.h>
     194             ^~~~~~~~~~~~~
     195    In file included from general.c:9:
  >> 196    ./_hypre_utilities.hpp:116:10: fatal error: 'rocsparse.h' file not found
     197    #include <rocsparse.h>
     198             ^~~~~~~~~~~~~
     199    In file included from handle.c:15:
  >> 200    ./_hypre_utilities.hpp:116:10: fatal error: 'rocsparse.h' file not found
     201    #include <rocsparse.h>
     202             ^~~~~~~~~~~~~
     203    1 error generated when compiling for gfx90a.
     204    1 error generated when compiling for gfx90a.
     205    In file included from int_array.c:9:
  >> 206    ./_hypre_utilities.hpp:116:10: fatal error: 'rocsparse.h' file not found
     207    #include <rocsparse.h>
     208             ^~~~~~~~~~~~~
     209    1 error generated when compiling for gfx90a.
  >> 210    make[1]: *** [../config/Makefile.config:66: general.obj] Error 1
     211    make[1]: *** Waiting for unfinished jobs....
  >> 212    make[1]: *** [../config/Makefile.config:66: device_utils.obj] Error 1
     213    1 error generated when compiling for gfx90a.
  >> 214    make[1]: *** [../config/Makefile.config:66: handle.obj] Error 1
  >> 215    make[1]: *** [../config/Makefile.config:66: int_array.obj] Error 1
     216    In file included from memory.c:15:
  >> 217    ./_hypre_utilities.hpp:116:10: fatal error: 'rocsparse.h' file not found
     218    #include <rocsparse.h>
     219             ^~~~~~~~~~~~~
     220    1 error generated when compiling for gfx90a.
  >> 221    make[1]: *** [../config/Makefile.config:66: memory.obj] Error 1
     222    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-hypre-2.25.0-6rsc2fdt7na5sqleojg3uug6klohx2f6/spack-src/src/utilities'
  >> 223    make: *** [Makefile:86: all] Error 1
```

### Error message

<details><summary>Error message</summary><pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.19.0.dev0 (9d5375a3ba150a3b216f06bc435b335b8f9bfc42)
* **Python:** 3.10.4
* **Platform:** linux-ubuntu22.04-zen2
* **Concretizer:** clingo

### Additional information

[spack-build-out.txt](https://github.com/spack/spack/files/9354620/spack-build-out.txt)
[spack-build-env.txt](https://github.com/spack/spack/files/9354621/spack-build-env.txt)

@balay @osborn9 @ulrikeyang @wspear

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/hypre/package.py
+++ b/var/spack/repos/builtin/packages/hypre/package.py
@@ -93,7 +93,9 @@ class Hypre(AutotoolsPackage, CudaPackage, ROCmPackage):
     depends_on("blas")
     depends_on("lapack")
     depends_on("superlu-dist", when="+superlu-dist+mpi")
-
+    depends_on("rocsparse", when="+rocm")
+    depends_on("rocthrust", when="+rocm")
+    depends_on("rocrand", when="+rocm")
     depends_on("umpire", when="+umpire")
     for sm_ in CudaPackage.cuda_arch_values:
         depends_on(
@@ -235,11 +237,16 @@ class Hypre(AutotoolsPackage, CudaPackage, ROCmPackage):
                 configure_args.append("--disable-cub")
 
         if "+rocm" in spec:
+            rocm_pkgs = ["rocsparse", "rocthrust", "rocprim", "rocrand"]
+            rocm_inc = ""
+            for pkg in rocm_pkgs:
+                rocm_inc += spec[pkg].headers.include_flags + " "
             configure_args.extend(
                 [
                     "--with-hip",
                     "--enable-rocrand",
                     "--enable-rocsparse",
+                    "--with-extra-CUFLAGS={0}".format(rocm_inc),
                 ]
             )
             rocm_arch_vals = spec.variants["amdgpu_target"].value

